### ISSUE TITLE
Is it possible to use on Linux?

### ISSUE THREAD

**Issue by philss** [wtsnz/obs-ios-camera-source#26]
Hi! Thank you for this project!

I want to stream programming sessions from my Ubuntu Linux 18.10, and I couldn't find instructions for how to build and install this plugin.
Is it possible to run on Linux?

PS: I already using on MacOS, but I wanna try on Linux.

**Comment by philss** [comment]
I tried and I couldn't make it work. Perhaps it's possible with some different compilation flags, but I gave up with OBS IOS Camera, and I could make work with other tools explained in this blog post: https://steemit.com/technology/@tech.ninja/tech-quickie-convert-your-ipad-iphone-into-a-good-wireless-webcam-in-linux

**Comment by dougg3** [comment]
Hey @philss, I saw you made this issue a while ago. I was able to get this plugin to compile on Ubuntu 18.04 by adding a few #includes in various places where I was seeing errors. I also had a problem at final link time because all of the libraries needed to be compiled with -fPIC. It seems to work! I ran into a quirk where sometimes I have to change between the front-facing and rear-facing camera in order to get the streaming to start working, but for the most part it seems to work okay with OBS 23.2.1.

Here is the commit to my branch that makes it work. The commit message has some compilation instructions. https://github.com/dougg3/obs-ios-camera-source/commit/2b6ed180f58c323b720d62b9226fbe86fb3387bd

After doing this, I manually created the directory structure for the plugin and put it in ~/.config/obs-studio/plugins:

```
obs-ios-camera-source/
obs-ios-camera-source/data
obs-ios-camera-source/data/locale
obs-ios-camera-source/data/locale/en-US.ini
obs-ios-camera-source/bin
obs-ios-camera-source/bin/64bit
obs-ios-camera-source/bin/64bit/obs-ios-camera-source.so
```

@wtsnz I'm not sure if these changes break the Mac OS X or Windows builds, or else I might be a little more confident about cleaning up the build process and trying to get it merged with your main project, if you're interested in supporting Linux at all. I also realize it could be a pain if you try to officially support Linux and a bunch of people run into problems...

**Comment by bkylerussell** [comment]
> Here is the commit to my branch that makes it work. The commit message has some compilation instructions. [dougg3@2b6ed18](https://github.com/dougg3/obs-ios-camera-source/commit/2b6ed180f58c323b720d62b9226fbe86fb3387bd)

This worked great for me on Ubuntu 18.04.  I had to also install libobs-dev so `FindLibObs.cmake` had the right pkg-config setup in place.  I was also missing libavcodec-dev, but once those were installed the commit referenced above worked great.  I'm not that familiar with OBS yet, but my plugin directory was `/usr/lib/x86_64-linux-gnu/obs-plugins`.

**Comment by jbberinger** [comment]
@dougg3 @bkylerussell  
Just tried to build on Ubuntu 19.10 from the commit instructions and got a bunch of CMake errors. Any ideas?
```
CMake Error at CMakeLists.txt:321 (target_compile_options):
  Cannot specify compile options for target "mbedcrypto" which is not built
  by this project.


CMake Error at CMakeLists.txt:322 (set_target_properties):
  set_target_properties Can not find target to add properties to:
  obs-ios-camera-source-win


CMake Error at CMakeLists.txt:323 (target_link_libraries):
  Cannot specify link libraries for target "obs-ios-camera-source-win" which
  is not built by this project.


-- Configuring incomplete, errors occurred!
```

**Comment by dougg3** [comment]
Hey @jbberinger, I don't think you checked out my branch -- make sure you do "git checkout linux-support" after cloning

**Comment by jbberinger** [comment]
@dougg3 Thanks, that was the problem! I'm a little confused about the final step where you created the directories, and where you got the .ini and .so files?

**Comment by dougg3** [comment]
You should be able to find those two files somewhere inside the build directory you created while following the build directions -- then all you need to do is arrange them in the directory structure indicated in my instructions and run OBS :-)

**Comment by philss** [comment]
Awesome, @dougg3! Thank you! ❤️ 
I will give a try someday.

**Comment by TSchattschneider** [comment]
> > Here is the commit to my branch that makes it work. The commit message has some compilation instructions. [dougg3@2b6ed18](https://github.com/dougg3/obs-ios-camera-source/commit/2b6ed180f58c323b720d62b9226fbe86fb3387bd)
> 
> This worked great for me on Ubuntu 18.04. I had to also install libobs-dev so `FindLibObs.cmake` had the right pkg-config setup in place. I was also missing libavcodec-dev, but once those were installed the commit referenced above worked great. I'm not that familiar with OBS yet, but my plugin directory was `/usr/lib/x86_64-linux-gnu/obs-plugins`.

While OBS, as packaged for ubuntu, will have its plugins in `/usr/lib/x86_64-linux-gnu/obs-plugins`, it will still read your custom plugins from `~/.config/obs-studio/plugins`, so you can put the plugin there (in `~/.config/...`) along with its locale `en-US.ini-file`, as @dougg3 stated. That way, it will be loaded by OBS and it will be properly labeled as "iOS Camera" instead of `OBSIOSCamera.Title`

**Comment by dougg3** [comment]
Just a heads up for future people who are trying to build the Linux version, I rearranged things a bit to help streamline the Linux build process and put all the relevant documentation in a single place. My fork's master branch now has the Linux fixes, along with an updated README.md with build/install instructions:

https://github.com/dougg3/obs-ios-camera-source

**Comment by maurojr86** [comment]
Hello @dougg3

I believe you are the plugin developer. Thanks for coding the OBS iOS Camera Source plugin. 

I'm totally new into Linux. I'm finding it super hard to follow up on the technical aspects of installing this build on a Mac Ubuntu 20.04.2

Would you be able to create a video or some instructions in more step by step simple terms please? 

I think that my main problem is when I get here: 

Manually install the plugin by copying relevant files into your OBS plugins directory (assuming 64-bit Linux):
mkdir -p ~/.config/obs-studio/plugins/obs-ios-camera-source/data/locale
mkdir -p ~/.config/obs-studio/plugins/obs-ios-camera-source/bin/64bit
cp ../data/locale/en-US.ini ~/.config/obs-studio/plugins/obs-ios-camera-source/data/locale/
cp obs-ios-camera-source.so ~/.config/obs-studio/plugins/obs-ios-camera-source/bin/64bit/ 

Thanks for understand.

Kindly regards.

MJ

**Comment by dougg3** [comment]
Hi @maurojr86,

I am not the main plugin developer but I did port it to Linux. Unfortunately I don’t have time to make a video or tutorial, but you should be able to run those commands in a terminal window after you compile it, and that should install the plugin. Alternatively since you are on Ubuntu 20.04, just look in the Downloads section of the README and follow those instructions: download the artifact and follow the unzip instructions. They are simpler.

**Comment by maurojr86** [comment]
Hey @dougg3 , I followed the instruction but in the end the terminal would not execute the codes but it did not tell me any error or information. I contacted Will from @wtsnz via Twitter, hoping he will guide me through it. Thanks anyways, eventually I will get there.
